### PR TITLE
set varlink timeout to 1 second

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
If I set the timeout to 0 will it just exit when the connection ends?

**Comment by baude** [comment]
no, 0 == hangs on indefinitely

File: cmd/podman/varlink.go [hunk 1]
```diff
@@ -52,7 +54,7 @@ func varlinkCmd(c *cli.Context) error {
 	}
 
 	// Run the varlink server at the given address
-	if err = service.Listen(args[0], 0); err != nil {
+	if err = service.Listen(args[0], 10*time.Second); err != nil {
```
Comment (rhatdan) [inline cmd/podman/varlink.go]: Lets add an --timeout option and allow to me specified in microseconds.

**Comment by rhatdan** [comment]
@kaysievers

How would this work with socket activation.  Basically I want the podman varlink executable to run for a very short period of time, and then be restarted via socket activation.  Do we risk missing calls?

**Comment by mheon** [comment]
In an ideal world, we want all requests serviced by a separate Podman process.

**Comment by kaysievers** [comment]
> How would this work with socket activation.

Started on-demand, exiting when the specified timeout is reached. The timeout starts counting when there are no active clients connections anymore. When exited, any new request will activate it again. Probably like you expect :)

File: cmd/podman/varlink.go (hunk 1 above)
Comment (baude) [reply cmd/podman/varlink.go]: what do you want the default to be then ?

**Comment by rhatdan** [comment]
Excellent @kaysievers.  Now we can it to timeout in a second and not have it running all the time.

**Comment by baude** [comment]
done

**Comment by jwhonce** [comment]
LGTM

File: cmd/podman/varlink.go [hunk 2]
```diff
@@ -27,10 +35,12 @@ var (
 )
 
 func varlinkCmd(c *cli.Context) error {
+
```
Comment (mheon) [inline cmd/podman/varlink.go]: Extra whitespace

**Comment by mheon** [comment]
One nit otherwise LGTM

**Comment by baude** [comment]
fixed up per comments ... merging

**Comment by baude** [comment]
@rh-atomic-bot r+

File: docs/podman-varlink.1.md [hunk 3]
```diff
@@ -6,34 +6,43 @@
...
 ## DESCRIPTION
-Starts the varlink service that allows varlink clients to interact with podman.
-<!--
-More will go here as the docs and api firm up.
--->
+Starts the varlink service that allows varlink clients to interact with podman.  This should generally be done
+with systemd.  See _Configuration_ below.
 
-**podman [GLOBAL OPTIONS] varlink **
+**podman [GLOBAL OPTIONS] [OPTIONS] varlink VARLINK_URI**
```
Comment (wking) [inline docs/podman-varlink.1.md]: Shouldn't this be:

    podman [GLOBAL OPTIONS] varlink [OPTIONS] VARLINK_URI

And shouldn't it go in the SYNOPSIS section above?

**Pull request closed without merging**